cwl/src/main/scala/cwl/InitialWorkDirRequirement.scala

```diff
@@ -58,6 +59,21 @@ object InitialWorkDirRequirement {
       def unapply(e: IwdrListingArrayEntry): Option[(Expression, Option[StringOrExpression], Boolean)] =
         e.select[ExpressionDirent].map(sd => (sd.entry, sd.entryname, sd.writableWithDefault))
     }
+    object FilePath {
+      def unapply(e: IwdrListingArrayEntry): Option[String] = e.select[File].flatMap(file => file.location)
+    }
+    object String {
+      def unapply(e: IwdrListingArrayEntry): Option[String] = e.select[StringOrExpression].flatMap(_.select[String])
+    }
+    object ECMAScriptExpression {
+      def unapply(e: IwdrListingArrayEntry): Option[ECMAScriptExpression] = for {
+        soe <- e.select[StringOrExpression]
+        expr <- soe.select[Expression]
+        ecmaScriptExpression <- expr.select[ECMAScriptExpression]
```

Comment:
Why does this only apply to `ECMAScriptExpression` and not `Expression`? According to the way I read [the spec](http://www.commonwl.org/v1.0/CommandLineTool.html#InitialWorkDirRequirement), `ECMAScriptFunction` should be valid wherever `ECMAScriptExpression` are valid.

Also, ToL, I haven't made up my mind yet whether I find `Poly1`s easier to read than these `unapply`s used in match statements.

Reply:
I'll address this in an iwdr_7 branch